### PR TITLE
fix(docs): docs targets

### PR DESCRIPTION
**Pull request by hickeyma**
**What this PR does / why we need it**:
Fix  the docs targets:
- `make docs`
- `make verify-docs`

partial #5397

**Special notes for your reviewer**:
`5048ed8` removed the pre-built docs from the code base. Following a discussion on slack with @adamreese @flynnduism, the docs targets are still required to generate the help docs for the web documentation. 

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by hickeyma** [comment]
@bacongobbler ok, thanks for the feedback. I will simplify the pr then as just remove the target as suggested.

**Comment by hickeyma** [comment]
@bacongobbler updated and ready for review again

File: docs/developers.md [hunk 1]
```diff
@@ -34,8 +34,7 @@ To run Helm locally, you can run `bin/helm`.
 
 ### Man pages
 
-Man pages and Markdown documentation are already pre-built in `docs/`. You may
-regenerate documentation using `make docs`.
+Man pages and Markdown documentation are already pre-built in `docs/`.
```
Comment (adamreese) [inline docs/developers.md]: This should be flipped to something like... Docs are not pre-built but you can generate them using `make docs`.

File: Makefile [hunk 2]
```diff
@@ -148,10 +144,6 @@ checksum:
 
 # ------------------------------------------------------------------------------
 
-.PHONY: docs
```
Comment (adamreese) [inline Makefile]: Is `make docs` no longer valid?

Comment (bacongobbler) [reply Makefile]: As far as I understand, yes. See above discussion: https://github.com/helm/helm/pull/5458#discussion_r269648921

Comment (hickeyma) [reply Makefile]: @adamreese @bacongobbler Lets discuss at the dev meeting today. I am confused npw whether the target is still needed or not.

File: docs/developers.md (hunk 1 above)
Comment (hickeyma) [reply docs/developers.md]: Done

File: Makefile (hunk 2 above)
Comment (hickeyma) [reply Makefile]: Following a discussion on slack with @adamreese @flynnduism, the docs targets are still required to generate the help docs for the web documentation.

**Comment by hickeyma** [comment]
Updated and ready for review again.

**Review by bacongobbler — APPROVED** [review]

**Comment by hickeyma** [comment]
Thanks @bacongobbler and @adamreese for reviews and merge.

**Pull request merged**